### PR TITLE
SLI and TaskSet Twitter Codebundles

### PR DESCRIPTION
**Pull request by stewartshea**
This PR adds a codebundle to query twitter for tweets based on a user handle, and keyword called SocialScrape (based on snscrape) that supports the twitter function above (though this can be extended to a wide range of social account use cases). 

The driving use behind this is an interim step to pull tweets from the GitBook status twitter account, at least until we have an understanding as to where their public status.io api endpoint is, but this might be useful for other users.

### CODE REVIEW

File: codebundles/twitter-query-tweets/sli.robot [hunk 1]
```diff
@@ -0,0 +1,42 @@
+*** Settings ***
+Metadata          Author    Shea Stewart
+Documentation     Queries Twitter to count amount of tweets within a specified time range for a specific user handle.
+Force Tags        Twitter    Social  tweet
+Suite Setup       Suite Initialization
+Library           RW.Core
+Library           RW.SocialScrape.SocialScrape
```
Comment (jon-funk) [inline codebundles/twitter-query-tweets/sli.robot:7]: Just a nitpick; Since you setup the __init__ file, you should be able to shorten this to `RW.SocialScrape`
and then use the method name as: `RW.SocialScrape.Twitter Scrape Handle`

File: codebundles/twitter-query-tweets/sli.robot [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    ...    default=0
+    Set Suite Variable    ${HANDLE}    ${HANDLE}
+    Set Suite Variable    ${MAX_TWEETS}    ${MAX_TWEETS}
+    Set Suite Variable    ${MAX_TWEET_AGE}    ${MAX_TWEET_AGE}
+    Set Suite Variable    ${MIN_TWEET_AGE}    ${MIN_TWEET_AGE}
+
+*** Tasks ***
+Query Twitter
+    ${rsp}=    RW.SocialScrape.SocialScrape.Twitter Scrape Handle    handle=${HANDLE}    maxTweets=${MAX_TWEETS}    max_tweet_age=${MAX_TWEET_AGE}    min_tweet_age=${MIN_TWEET_AGE}
+    ${metric} =    Get Length    ${rsp}
```
Comment (jon-funk) [inline codebundles/twitter-query-tweets/sli.robot:39]: `${metric} =` got an extra space hiding in there

File: libraries/RW/SocialScrape/SocialScrape.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+from RW import platform
+import snscrape.modules.twitter as sntwitter
+import pandas as pd
+
+class SocialScrape: 
+    """
+    Twitter Scraper keyword library
+    Uses https://github.com/JustAnotherArchivist/snscrape
+    """
+
```
Comment (jon-funk) [inline libraries/RW/SocialScrape/SocialScrape.py:26]: We typically add an explicit:
`ROBOT_LIBRARY_SCOPE = "GLOBAL"`
in class stanzas to indicate their scoping within robot suites. This indicates to robotframework to maintain 1 instance of the class within the suite. It won't make a functional difference to what you've written but it's a good maintainability addition.

**Review by jon-funk — CHANGES_REQUESTED** [review]
Really cool addition to the codecollection, I'm excited to see this in action. I've added a few nitpicky comments.

File: codebundles/twitter-query-tweets/sli.robot (hunk 1 above)
Comment (stewartshea) [reply codebundles/twitter-query-tweets/sli.robot]: 👍  Thanks for the review and helpful details!

File: codebundles/twitter-query-tweets/sli.robot (hunk 2 above)
Comment (stewartshea) [reply codebundles/twitter-query-tweets/sli.robot]: 👍 eagle eye :) Thanks!

**Comment by stewartshea** [comment]
@jon-funk the latest commit to the main branch has removed `https://backend-services.dev.project-468.com/pypi/platform/rw.core/RW.Core-0.1.1.tar.gz` from requirements.txt (which is the cause of the conflict). Was this intentional? If so, I can rectify it on my side.

**Comment by jon-funk** [comment]
Yes - it was vestigial from the prior repo. That package is included in the coderun builds by default during the kaniko build and otherwise should only be included in devtools builds.

**Review by jon-funk — APPROVED** [review]
